src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,151 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.stream.Collectors;
+
+/**
+ * Tasklist class containing a list of tasks.
+ */
+public class TaskList {
+    private static ArrayList<Task> taskList;
+
+    public TaskList(ArrayList<Task> taskList) throws DukeException {
+        try {
+            this.taskList = taskList;
+        } catch (Exception e) {
```

Comment:
I am not sure if the try-catch here is really necessary. From what I understand, if any error did occur (possibly when loading the list), it should be handled by either the `Storage` class or the `Parser` class? Perhaps you might want to change the `Exception` to be more specific so as to avoid catching all exceptions when debugging?